### PR TITLE
perf improvements and benchmarking

### PR DESCRIPTION
**Pull request by abbasc52**
- Made LocalParams and ErrorMessage formatting optional via ReSettings
- Major performance improvement
	- 25% improvement from previous version
	- Upto 35% improvement by disabling optional features

### CODE REVIEW

File: src/RulesEngine/Models/Rule.cs [hunk 1]
```diff
@@ -17,14 +17,6 @@ namespace RulesEngine.Models
...
     {
         public string RuleName { get; set; }
-
-        /// <summary>
-        /// Gets or sets the custom property or tags of the rule.
-        /// </summary>
-        /// <value>
-        /// The properties of the rule.
-        /// </value>
-        public Dictionary<string, object> Properties { get; set; }
```
Comment (abbasc52) [inline src/RulesEngine/Models/Rule.cs:27]: revert this change

**Review by dishantmunjal — APPROVED** [review]
